docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,213 @@
+# Contributing to Devworkspace-Operator
+
+Hello there! thank you for choosing to the contributing to devfile/devworkspace-operator. Navigate through the following to understand more about contributing here.
+
+- [Contributing to Devworkspace-Operator](#contributing-to-devworkspace-operator)
+- [Before You Get Started](#before-you-get-started)
+  - [Code of Conduct](#code-of-conduct)
+  - [For Newcomers](#for-newcomers)
+- [How to Contribute](#how-to-contribute)
+  - [Set up your Local Development Environment](#set-up-your-local-development-environment)
+  - [Testing Changes](#testing-changes)
+  - [Signing-off on Commits](#signing-off-on-commits)
+    (#testing-your-changes)
+  - [Signing-off on Commits](#signing-off-on-commits)
+
+# Before You Get Started
+
+## Code of Conduct
+
+## For Newcomers
+
+# How to Contribute
+
+<!--
+## Prerequisites
+
+Make sure you have the following prerequisites installed on your operating system before you start contributing:
+
+- [Nodejs and npm](https://nodejs.org/en/)
+
+  To verify run:
+
+  ```
+  node -v
+  ```
+
+  ```
+  npm -v
+  ```
+
+- [Gatsby.js](https://www.gatsbyjs.com/)
+
+  To verify run:
+
+  ```
+  gatsby --version
+  ```
+
+**Note:** If you're on a _Windows environment_ then it is highly recommended that you install [Windows Subsystem for Linux (WSL)](https://docs.microsoft.com/en-us/windows/wsl/install) both for performance and ease of use. Refer to the [documentation](https://docs.microsoft.com/en-us/windows/dev-environment/javascript/gatsby-on-wsl) for the installation of _Gatsby.js on WSL_. -->
+
+## Set up your Local Development Environment
+
+Follow the following instructions to start contributing.
+
+**1.** Fork [this](https://github.com/devfile/devworkspace-operator) repository.
```

Comment:
> @kernelpanic77 looking great so far sunglasses In addition to @amisevsk's review, left some minor comments and potential suggestions.
> 
> After these comments are addressed, it might be worth adding a section on developing the webhook. IIRC, this process involves:

Thanks @AObuchow. This makes sense, I'll add a specific sub section for testing and developing webhooks in the "testing your changes" section.
 
> 1. Make a change to the webhook
> 2. Ensure the DWO_IMG environment variable points to your container image repository, eg. `export DWO_IMG=quay.io/aobuchow/dwo-webhook:next`
> 3. Run `make docker restart` (assuming DWO is already deployed to the cluster, otherwise `make docker install)`
> 4. Wait for the webhook deployment to update with your image that contains your latest changes.
> 
> There may be a better way to develop the webhook, but this is generally the flow that I've performed.
I have not contributed to developing webhooks yet, so this looks right to me. @amisevsk @ibuziuk would you like to add any steps to the process specified by @AObuchow.